File: 2574. left-and-right-sum-differences.java
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int rightSum = 0;

        for (int num : nums) {
            rightSum += num;
        }

        int leftSum = 0;
        int[] result = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            rightSum -= nums[i];
            result[i] = Math.abs(leftSum - rightSum);
            leftSum += nums[i];
        }

        return result;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        int[] nums1 = {10,4,8,3};
        int[] output1 = sol.leftRightDifference(nums1);
        System.out.println("Example 1: nums=[10,4,8,3]");
        System.out.print("Output: [");
        for(int i = 0; i < output1.length; i++) {
            if(i > 0) System.out.print(",");
            System.out.print(output1[i]);
        }
        System.out.println("]");
        System.out.println("Expected: [15,1,11,22]");
        System.out.println();
        
        // Example 2
        int[] nums2 = {1};
        int[] output2 = sol.leftRightDifference(nums2);
        System.out.println("Example 2: nums=[1]");
        System.out.print("Output: [");
        for(int i = 0; i < output2.length; i++) {
            if(i > 0) System.out.print(",");
            System.out.print(output2[i]);
        }
        System.out.println("]");
        System.out.println("Expected: [0]");
    }
}

File: 2574. left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        right_sum = sum(nums)

        left_sum = 0
        result = []

        for num in nums:
            right_sum -= num
            result.append(abs(left_sum - right_sum))
            left_sum += num

        return result

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    nums1 = [10,4,8,3]
    output1 = sol.leftRightDifference(nums1)
    print(f"Example 1: nums={nums1}")
    print(f"Output: {output1}")
    print(f"Expected: [15,1,11,22]")
    print()
    
    # Example 2
    nums2 = [1]
    output2 = sol.leftRightDifference(nums2)
    print(f"Example 2: nums={nums2}")
    print(f"Output: {output2}")
    print(f"Expected: [0]")